core/arch/arm/tee/entry_fast.c

```diff
@@ -217,6 +239,69 @@ static void get_async_notif_value(struct thread_smc_args *args)
 		args->a2 |= OPTEE_SMC_ASYNC_NOTIF_PENDING;
 }
 
+static void get_pending_notif(struct thread_smc_args *args)
+{
+	bool it_valid = false;
+	bool it_pending = false;
+	int it[5] = { -1, -1, -1, -1, -1 };
+	ssize_t cnt = 0;
+
+	do {
+		uint32_t it_num  = notif_it_get_value(&it_valid, &it_pending);
```

Comment:
I'd rather have one single function to dig out all the needed information, something like:
``` C
/*
 * Report pending interrupts and asynchronous notification values.
 * @do_bottom_half: set to true if NOTIF_VALUE_DO_BOTTOM_HALF is pending, and
 *                  clears it as delivered.
 * @async_value:    set to true if any other asynchronous notification value
 *                  is pending.
 * @it_nums:        array of size *@it_count to report pending interrupts
 * @it_count:       at most @it_count interrupts are reported in @it_nums,
 *                  @it_count is updated with the number of reported interrupts.
 */
void notif_get_pending(bool *do_bottom_half, bool *async_value,
                       uint16_t *it_nums, size_t *it_count);
```

Reply:
ok, i'll do that.